apps/email/js/date.js

```diff
@@ -1,14 +1,35 @@
 'use strict';
 define(function(require) {
-  var mozL10n = require('l10n!');
+  const mozIntl = require('moz_intl');
+  var shortRelativeDateFmt = mozIntl._gaia.RelativeDate(navigator.languages, {
+    style: 'short'
+  });
+  var longRelativeDateFmt = mozIntl._gaia.RelativeDate(navigator.languages, {
+    style: 'long'
+  });
+
+  function updateRelativeDateFormatters() {
+    shortRelativeDateFmt = mozIntl._gaia.RelativeDate(navigator.languages, {
+      style: 'short'
+    });
+    longRelativeDateFmt = mozIntl._gaia.RelativeDate(navigator.languages, {
+      style: 'long'
+    });
+  }
+
+  window.addEventListener('languagechange', updateRelativeDateFormatters);
```

Comment:
It would be good to reduce the calls to RelativeDate, just for ease of updating later. Maybe just declare the shortRelativeDateFmt and longRelativeDateFmt variables, then call updateRelativeDateFormatters right away to set up the initial values, then register the languagechange listener.